### ISSUE TITLE
"...has been blocked from an action..."

### ISSUE THREAD

**Issue by PeeWee-Sweden** [SLOKnightfall/BetterWardrobe#397]
Version 4.14.1   ....trying to upgrade an item at Cuzolth in Loamm

**Comment by SLOKnightfall** [comment]
Please leave a taint log if you could.  Right now being able to review taint logs from people is the only way to fix the issue since I'm not experiencing it and the logs are the only way to try to track it back to the source.  https://wowpedia.fandom.com/wiki/CVar_taintLog

**Comment by PeeWee-Sweden** [comment]
Ok, I'll have that in mind! I have enabled taint-logging and will include it if it happens again.

**Comment by Nuthar** [comment]
Just wanted to contribute my own findings because I've been fighting the "blocked from an action" issues since I got this addon, which drives me nuts because I love its features.

I notice it happens after I learn a new appearance. I've tested repeatedly and I don't get the error at all when I haven't learned a new appearance, and the moment I do unlock a new one, I'm unable to right-click and use any items in my bags.

The taint log entries look like:

10/19 20:52:28.730  Interface/FrameXML/ContainerFrame.lua:1604 GetBagID()
10/19 20:52:28.730  An action was blocked because of taint from BetterWardrobe - 
10/19 20:52:28.730      UseContainerItem()
10/19 20:52:28.730      Interface/FrameXML/ContainerFrame.lua:1424 ContainerFrameItemButton_OnClick()
10/19 20:52:28.730      Interface/FrameXML/ContainerFrame.lua:1476

I'm presently using version 4.15.1 but this has persisted for at least a couple months and I've been updating over time. 

I tried using this addon as the only enabled one and didn't get the taint errors - but then tried gradually rotating in other addons, and it seems like a good portion of them, when loaded as my only other enabled addon alongside BetterWardrobe, will prompt the taint errors to start appearing again. Disabling BetterWardrobe with any combination of other addons enabled results in the taint error ceasing.

Thanks!

**Comment by SLOKnightfall** [comment]
Thanks for the logs and notes on when it appears to happen.  That will help me narrow things down.

**Comment by SLOKnightfall** [comment]
@Nuthar Could you run a test using Better Wardrobe and a few of the addons that you notice the issue with by buying a cheep mog on the ah to trigger your taint issue and then attach the entire log file to a comment.  That might help me backtrace the source a bit clearer.

**Comment by Nuthar** [comment]
I've been trying a few variations of the use case that's been triggering the errors to try and narrow down the conditions as much as I can.

It turns out that I **don't** actually need any other addons enabled for the error to trigger; I must have just not done one of the key steps that triggers it when I had originally gotten the impression that combining with other addons made a difference. To be clear, I get the taint error using the following steps with **only** BetterWardrobe and BW SourceData enabled in my addons list.

I get the error after doing the following steps:

- Learn new appearance from equipping an item
- Open collection list
- Ensure that both the collection list and any bags are then closed
- Open bags once more
- Attempt to use an item in my bags (test case was with a Flight Master's Whistle, but seems to affect all items that are used on right-click)

If skipping the collection window step, it seems the error does not appear.
If I've kept my bags open the whole time, I can still use items without the error appearing.
Only after viewing the collection window and newly opening my bags does the error seem to appear.

I produced a series of Taint log files by using the same steps, trying out variations of each step to eliminate the possibility that the steps had to be performed in a specific manner (e.g. clicking buttons vs. keybinds to open collection/bags doesn't seem to make a difference)

[taint.log](https://github.com/SLOKnightfall/BetterWardrobe/files/13064965/taint.log)
[taint 2.log](https://github.com/SLOKnightfall/BetterWardrobe/files/13064968/taint.2.log)
[taint 3.log](https://github.com/SLOKnightfall/BetterWardrobe/files/13064970/taint.3.log)
[taint 4.log](https://github.com/SLOKnightfall/BetterWardrobe/files/13064971/taint.4.log)
[taint 5.log](https://github.com/SLOKnightfall/BetterWardrobe/files/13064973/taint.5.log)

**Comment by SLOKnightfall** [comment]
@Nuthar Thank you, thank you, and thank you once again.  These detailed notes and experiments really help vs someone just posting the blocked message.  Especially the steps that cause you to get the error since I can try to follow the exact steps and see if I can reproduce.

**Comment by SLOKnightfall** [comment]
Fingers crossed that I'm 'might' have fixed the taint issue or at least reduced it.  I reworked a bunch more stuff and reduced the Execution tainted portions of the logs.  I followed the steps from above and there were no blocked calls from the bags or toy box.

**Comment by Nuthar** [comment]
I think you may have indeed cracked the issue! When performing the same steps as my old test scenario, I am not getting error messages when using items from my bags. Very excited, thank you so much for your work on this and for following up!

I noticed what appears to be a new issue that popped up between versions - when in the Sets or Extra tab, mousing over the "Hide Armor Slots" button will pop up the tooltip, but it's larger than it was previously and persists indefinitely once it appears; mousing off of the button and/or closing the Appearances interface don't seem to make it go away.

Thought I'd bring it up here in case it was tied directly to any of the changes made toward fixing this issue, but happy to submit as a separate issue if it would help!

**Comment by SLOKnightfall** [comment]
Oooooooo!  That is what I was hoping to hear, minus the tooltip issue which should be fixed in the next update :)
